src/yadism/coefficient_functions/heavy/g1_nc.py

```diff
@@ -0,0 +1,154 @@
+        :eqref:`D.14` of :cite:`felix-thesis`.
+        """
+
+        def cg(z, _args):
+            if self.is_below_threshold(z):
+                return 0.0
+            return (
+                self._FHprefactor
+                / z
+                * (4.0 * np.pi) ** 2
+                * (
+                    LeProHQ.cg1("x2g1", "VV", self._xi, self._eta(z))
+                    + LeProHQ.cgBar1("x2g1", "VV", self._xi, self._eta(z))
+                    * np.log(self._xi)
+                )
+            )
+
+        return RSL(cg)
+
+
+class GluonAA(GluonVV):
+    def NLO(self):
+        """
+        |ref| implements :eqref:`D.6`, :cite:`felix-thesis`.
+        """
+
+        def cg(z, _args):
+            if self.is_below_threshold(z):
+                return 0.0
+            return (
+                self._FHprefactor
+                / z
+                * LeProHQ.cg0("x2g1", "AA", self._xi, self._eta(z))
+            )
+
+        return RSL(cg)
+
+    def NNLO(self):
+        """
+        |ref| implements |NNLO| (heavy) gluon coefficient function,
+        :eqref:`D.14` of :cite:`felix-thesis`.
+        """
+
+        def cg(z, _args):
+            if self.is_below_threshold(z):
+                return 0.0
+            return (
+                self._FHprefactor
+                / z
+                * (4.0 * np.pi) ** 2
+                * (
+                    LeProHQ.cg1("x2g1", "AA", self._xi, self._eta(z))
+                    + LeProHQ.cgBar1("x2g1", "AA", self._xi, self._eta(z))
+                    * np.log(self._xi)
+                )
+            )
+
+        return RSL(cg)
+
+
+class SingletVV(pc.NeutralCurrentBase):
+    def NNLO(self):
+        """
+        |ref| implements |NNLO| (heavy) singlet coefficient function,
+        :eqref:`D.43` of :cite:`felix-thesis`.
+        """
+
+        def cq(z, _args):
+            if self.is_below_threshold(z):
+                return 0.0
+            return (
+                self._FHprefactor
+                / z
+                * (4.0 * np.pi) ** 2
+                * (
+                    LeProHQ.cq1("x2g1", "VV", self._xi, self._eta(z))
+                    + LeProHQ.cqBarF1("x2g1", "VV", self._xi, self._eta(z))
+                    * np.log(self._xi)
+                )
+            )
+
+        return RSL(cq)
+
+
+class SingletAA(pc.NeutralCurrentBase):
+    def NNLO(self):
+        """
+        |ref| implements |NNLO| (heavy) singlet coefficient function,
+        :eqref:`D.43` of :cite:`felix-thesis`.
+        """
+
+        def cq(z, _args):
+            if self.is_below_threshold(z):
+                return 0.0
+            return (
+                self._FHprefactor
+                / z
+                * (4.0 * np.pi) ** 2
+                * (
+                    LeProHQ.cq1("x2g1", "AA", self._xi, self._eta(z))
+                    + LeProHQ.cqBarF1("x2g1", "AA", self._xi, self._eta(z))
+                    * np.log(self._xi)
+                )
+            )
```

Comment:
Mh? Singlet (in my thesis called Bethe-Heitler) is $c_q$ and this is genuine new and not related to the massless coefficient functions - more over also here there is $c_q^{AA}\neq c_q^{VV}$ because the photon is (same as in the gluonic case and unlike in the non-singlet case) coupling to the massive quark (and so $\gamma_5$ is messing up)

Reply:
Sorry, I don't know what happened when adding the review but I meant to refer to the following passage:

https://github.com/NNPDF/yadism/blob/902a6c6ed9c526bee753b7f0f3d2589a94154f40/src/yadism/coefficient_functions/heavy/g1_nc.py#L137-L154

Reply:
okay that makes more sense ... [my comment from above](https://github.com/NNPDF/yadism/pull/185#discussion_r1168656051) still holds

Reply:
if you trace back the equation to https://github.com/felixhekhorn/LeProHQ/blob/main/LeProHQpy/src/LeProHQ/raw/dq1.py you see that it refers to the right equation for the coulomb Coefficient function as in the paper for the massive case, wiht the relations between cf's holding. so this is okay i think or maybe I am misunderstanding the issue.